tests/unit/Deserializers/DeserializerBaseTest.php

```diff
@@ -57,8 +57,15 @@ public function testIsDeserializerForReturnsFalse( $nonSerializable ) {
 	 * @dataProvider nonDeserializableProvider
 	 */
 	public function testSerializeThrowsUnsupportedObjectException( $nonSerializable ) {
-		$this->setExpectedException( 'Deserializers\Exceptions\DeserializationException' );
-		$this->buildDeserializer()->deserialize( $nonSerializable );
+		$deserializer = $this->buildDeserializer();
+
+		if ( $deserializer instanceof DispatchableDeserializer ) {
+			$this->setExpectedException( 'Deserializers\Exceptions\DeserializationException' );
+			$deserializer->deserialize( $nonSerializable );
+		}
+		else {
+			$this->assertTrue( true );
```

Comment:
-2. This disables all test cases in all `nonDeserializableProvider`s for all classes that are not dispatchable. But the fact that a deserializer throws exceptions does not depend on this.

Please don't do anything, I'm preparing a patch right now.